client/blocks/site-preview/index.jsx

```diff
@@ -1,35 +1,119 @@
 /** @format */
-
 /**
  * External dependencies
  */
-
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
+import debugFactory from 'debug';
 
 /**
  * Internal dependencies
  */
-import UrlPreview from 'blocks/url-preview';
+import { closePreview } from 'state/ui/preview/actions';
+import { getPreviewSite, getPreviewSiteId, getPreviewUrl } from 'state/ui/preview/selectors';
+import { getSiteOption, getSiteSlug } from 'state/sites/selectors';
 import { getCurrentLayoutFocus } from 'state/ui/layout-focus/selectors';
+import { addQueryArgs } from 'lib/route';
+import isDomainOnlySite from 'state/selectors/is-domain-only-site';
+import WebPreview from 'components/web-preview';
+
+const debug = debugFactory( 'calypso:design-preview' );
 
-class SitePreview extends Component {
+class UrlPreview extends Component {
 	static propTypes = {
+		className: PropTypes.string,
 		showPreview: PropTypes.bool,
+		previewUrl: PropTypes.string,
+		selectedSite: PropTypes.object,
+		selectedSiteId: PropTypes.number,
+		selectedSiteNonce: PropTypes.string,
+		selectedSiteUrl: PropTypes.string,
+		closePreview: PropTypes.func.isRequired,
 	};
 
-	static defaultProps = {
-		showPreview: false,
-	};
+	constructor( props ) {
+		super( props );
+		this.state = { previewCount: 0 };
+		this.previewCounter = 0;
+		this.onClosePreview = this.onClosePreview.bind( this );
```

Comment:
could use the bound method syntax instead?

Reply:
I assumed that @jsnajdr was trying to change as little of the implementation as possible when moving it.  This syntax is the exact same as existed before in the original component. I think that is a good thing for the purposes of this PR

Reply:
It's not very obvious from the diff, but I just moved 99% of the code from `blocks/url-preview` to `blocks/site-preview`.

But now when the PR is almost fully reviewed, I could push a few commits that bring up the component to contemporary code standards.